packages/generators/src/init-generator.ts

```diff
@@ -86,4 +107,24 @@ export default class InitGenerator extends CustomGenerator {
         this.utils.logger.log(`${blue('ℹ INFO ')} Initialising project...`);
         handlers[this.template].generate(this);
     }
+
+    public end(): void {
+        // Prettify configuration file if possible
+        try {
+            // eslint-disable-next-line node/no-extraneous-require, @typescript-eslint/no-var-requires
+            const prettier = require('prettier');
+            const source = readFileSync(this.configurationPath, { encoding: 'utf8' });
+            const formattedSource = prettier.format(source, {
+                singleQuote: true,
+                trailingComma: 'all',
+                printWidth: 140,
+                tabWidth: 4,
+                parser: 'babel',
+            });
+            writeFileSync(this.configurationPath, formattedSource);
+        } catch (err) {
+            this.utils.logger.log(`${yellow(`⚠ Generated configuration may not be properly formatted as prettier is not installed.`)}`);
+            return;
+        }
+    }
```

Comment:
Maybe we avoid extra reading file? But I agree, it is good idea

Reply:
I did that because if user opts to install prettier, we can only prettify file after `installing` context of yeoman. Only context that comes after `installing` in yeoman life-cycle is `end`. I considered above solution better as compared to installing prettier earlier using child process because we are not wasting time running npm install twice.

Reply:
If you have a suggestion let me know.

Reply:
Maybe `yeoman` have hooks before writing files? I don't familiar, so maybe you approach is good, just want check it

Reply:
> Maybe yeoman have hooks/callbacks before writing files?

Nothing I am aware of, Files are written first, then packages are installed (https://yeoman.io/authoring/running-context.html). No way other than installing prettier using child process before writting.